_shared/project/docker-compose.yml

```diff
@@ -1,8 +1,19 @@
-version: "3.9"
 services:
-  web:
-    build: .
+  {% if cookiecutter.get("db") == "yes" %}
+  postgres:
+    image: postgres:{{ cookiecutter.get("__postgres_version", "11.5-alpine") }}
     ports:
-      - "8000:8000"
-  redis:
-    image: "redis:alpine"
+      - "127.0.0.1:{{ cookiecutter['__db_port'] }}:5432"
+    healthcheck:
+      test: ["CMD", "pg_isready", "-U", "postgres"]
+      interval: 1s
+    {% if include_exists("docker-compose/postgres.yml") %}
+      {{- include("docker-compose/postgres.yml", indent=4) -}}
+    {% endif %}
+  {% endif %}
+  {% if include_exists("docker-compose/services.yml") %}
+    {{- include("docker-compose/services.yml", indent=2) -}}
+  {% endif %}
```

Comment:
I think they're at least potentially necessary:

* For each service definition that the cookiecutter supports in `docker-compose.yml` (currently only `postgres`) we're gonna want an include for adding arbitrary lines to the bottom of that service definition (the lines being indented so as to be part of the Postgres service definition) so that the project can customise the service definition. Not sure whether any of our projects actually _will_ do this for the Postgres service, but the potential need is obvious. If we add special knowledge about any other services in future (e.g. `elasticsearch` or whatever) then each of those will get its own include too
* You want an include for the bottom of the `services` section of the `docker-compose.yml` file (the included lines being indented so as to be part of the `services` section). This is how projects can add their own arbitrary service definitions that the cookiecutter knows nothing about (currently needed for any services other than Postgres)
* You want a final include for the very bottom of the `docker-compose.yml` file (not indented, not part of the `services` section). For example h and LMS will use this last include to add some networking stuff to `docker-compose.yml`.